scrapy/utils/url.py

```diff
@@ -114,10 +115,10 @@ def escape_ajax(url):
 
 def add_http_if_no_scheme(url):
     """Add http as the default scheme if it is missing from the url."""
-    if url.startswith('//'):
-        url = 'http:' + url
-        return url
-    parser = parse_url(url)
-    if not parser.scheme or not parser.netloc:
-        url = 'http://' + url
+    match = re.match(r"^\w+://", url, flags=re.I)
+    parts = urlparse(url)
+    if not match:
+        scheme = "http:" if parts.netloc else "http://"
+        url = scheme + url
```

Comment:
`add_http_if_no_scheme` function changed, but its tests were not - do you know if there are any differences in behaviour in old and new version? Not sure if it matters, but old version looks more optimized. `parse_url` -> `urlparse` is a reasonable change.